internal/txapp/txapp.go

```diff
@@ -667,3 +712,18 @@ func logErr(l log.Logger, err error) {
 		l.Error("error committing/rolling back transaction", zap.Error(err))
 	}
 }
+
+func computeEmptyVoteBodyTxSize(chainID string) (int64, error) {
+	// Create a transaction with an empty payload to measure the fixed size without the payload.
+	tx, err := transactions.CreateTransaction(&transactions.ValidatorVoteBodies{
+		Events: []*types.VotableEvent{},
+	}, chainID, 0)
+	if err != nil {
+		return 0, err
+	}
+	sz, err := tx.MarshalBinary()
+	if err != nil {
+		return 0, err
+	}
+	return int64(len(sz)), nil
+}
```

Comment:
Let's be more conservative/realistic with a non-zero nonce (zero is encoded shorter by RLP), and a non-zero fee (larger big.Int encoding):

```go
func computeEmptyVoteBodyTxSize(chainID string) (int64, error) {
	// Create a transaction with an empty payload to measure the fixed size without the payload.
	tx, err := transactions.CreateTransaction(&transactions.ValidatorVoteBodies{
		Events: []*types.VotableEvent{},
	}, chainID, 1<<63)
	if err != nil {
		return 0, err
	}
	tx.Body.Fee, _ = big.NewInt(0).SetString("987654000000000000000000000000000", 10)
	sz, err := tx.MarshalBinary()
	if err != nil {
		return 0, err
	}
	return int64(len(sz)), nil
}
```

```go
func Test_computeEmptyVoteBodyTxSize(t *testing.T) {
	tests := []struct {
		name    string
		chainID string
		want    int64
		wantErr bool
	}{
		{"asdf", "asdf", 72, false},
		{"abcdefghijklmnop", "abcdefghijklmnop", 84, false},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := computeEmptyVoteBodyTxSize(tt.chainID)
			if (err != nil) != tt.wantErr {
				t.Errorf("computeEmptyVoteBodyTxSize() error = %v, wantErr %v", err, tt.wantErr)
				return
			}
			if got != tt.want {
				t.Errorf("computeEmptyVoteBodyTxSize() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

Reply:
Makes sense.